contracts/TreasuryManager.sol

```diff
@@ -280,7 +253,7 @@ contract TreasuryManager is
         uint256 wethAmount,
         uint256 noteAmount,
         uint256 minBPT
-    ) external onlyManager {
+    ) external onlyManager onlyOnMainnet {
```

Comment:
We had this snapshot vote pass here:
https://snapshot.org/#/notional.eth/proposal/0x491d4de954b2e27a29fddf57f8ecfc593dcf24765db6a568e6e6195634f1652d

So I'd like to update this method as follows:

1. Add a new governance storage parameter called `noteBurnPercent` settable by the owner
2. Calculate the `wethForNOTEBurn` amount as  `wethAmount * noteBurnPercent`.
3. Execute an `EXACT_IN_SINGLE` or `EXACT_IN_BATCH` trade where the sellToken = WETH, the buyToken = NOTE and the amount is set to wethForNOTEBurn via the TradingModule. You can use the DexId 0x for this trade.
4. Use the returned `amountBought` from the trade above and burn the NOTE by sending it to `address(0)`
5. Use the remaining wethAmount `(wethAmount - wethForNOTEBurn)` for the Balancer pool reinvestment. If any noteAmount is passed in it will go to the balancer pool reinvestment.